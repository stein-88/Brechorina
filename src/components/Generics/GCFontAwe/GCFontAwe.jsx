import React from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import * as solid from '@fortawesome/free-solid-svg-icons'
import * as regular from '@fortawesome/free-regular-svg-icons'
import * as brands from '@fortawesome/free-brands-svg-icons'
import { UpperFirst } from '@utils'

const GCFontAwe = ({ customClass, nameIco, type = 'regular' }) => {
    const allTypes = { regular, solid, brands }
    if (!nameIco || !allTypes?.[type]) return null
    const finalName = nameIco.split('-').map((cv, ind) => (ind === 0 ? cv : UpperFirst(cv))).join('')
    const finalIcon = allTypes?.[type]?.[finalName]
    if (!finalIcon) return null
    return <FontAwesomeIcon icon={finalIcon} className={customClass || ''} />
}

export default GCFontAwe
