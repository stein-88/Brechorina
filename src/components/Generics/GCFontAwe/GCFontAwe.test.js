import React from 'react'
import renderer from 'react-test-renderer'
import GCFontAwe from './GCFontAwe'

describe('Test on <GCFontAwe /> component', () => {
    test('Should not render without the props', () => {
        const componentNoprops = renderer.create(<GCFontAwe />)
        const treeNoprops = componentNoprops.toJSON()
        expect(treeNoprops).toBeNull()
    })

    test('Should render with minimum props', () => {
        const componentNoprops = renderer.create(<GCFontAwe nameIco="fa-circle" />)
        const treeNoprops = componentNoprops.toJSON()
        expect(treeNoprops).toMatchSnapshot()
    })

    test('Should not render without finalIcon', () => {
        const componentNoprops = renderer.create(<GCFontAwe nameIco="batata" />)
        const treeNoprops = componentNoprops.toJSON()
        expect(treeNoprops).toBeNull()
    })
})
