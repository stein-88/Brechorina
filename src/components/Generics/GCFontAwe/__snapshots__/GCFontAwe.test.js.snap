// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <GCFontAwe /> component Should render with minimum props 1`] = `
<svg
  aria-hidden="true"
  className="svg-inline--fa fa-circle "
  data-icon="circle"
  data-prefix="far"
  focusable="false"
  role="img"
  style={Object {}}
  viewBox="0 0 512 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M464 256A208 208 0 1 0 48 256a208 208 0 1 0 416 0zM0 256a256 256 0 1 1 512 0A256 256 0 1 1 0 256z"
    fill="currentColor"
    style={Object {}}
  />
</svg>
`;
